src/libsyntax/attr/mod.rs

```diff
@@ -347,16 +347,17 @@ impl Attribute {
 
 pub fn mk_name_value_item_str(ident: Ident, value: Spanned<Symbol>) -> MetaItem {
     let lit_kind = LitKind::Str(value.node, ast::StrStyle::Cooked);
-    mk_name_value_item(ident.span.to(value.span), ident, lit_kind, value.span)
+    let pos = ident.span.to(value.span);
+    mk_name_value_item(ident.with_span_pos(pos), lit_kind, value.span)
```

Comment:
I'd rather move the span concatenation logic into `mk_name_value_item` and avoid changing the ident span.